protocol/app/ante/timestampnonce.go

```diff
@@ -0,0 +1,55 @@
+package ante
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	accountpluskeeper "github.com/dydxprotocol/v4-chain/protocol/x/accountplus/keeper"
+)
+
+// TODO: combine increment sequence and sequence verification into one decorator
+// https://github.com/cosmos/cosmos-sdk/pull/18817
+type IncrementSequenceDecorator struct {
+	ak ante.AccountKeeper
+}
+
+func NewIncrementSequenceDecorator(ak ante.AccountKeeper) IncrementSequenceDecorator {
+	return IncrementSequenceDecorator{
+		ak: ak,
+	}
+}
+
+func (isd IncrementSequenceDecorator) AnteHandle(
+	ctx sdk.Context,
+	tx sdk.Tx,
+	simulate bool,
+	next sdk.AnteHandler,
+) (sdk.Context, error) {
+	sigTx, ok := tx.(authsigning.SigVerifiableTx)
+	if !ok {
+		return ctx, errorsmod.Wrap(sdkerrors.ErrTxDecode, "invalid transaction type")
+	}
+
+	signatures, err := sigTx.GetSignaturesV2()
+	if err != nil {
+		return sdk.Context{}, err
+	}
+
+	for _, signature := range signatures {
+		if accountpluskeeper.IsTimestampNonce(signature.Sequence) {
```

Comment:
We have a very specific use case for timestamp nonce feature, and it seems to me the following will never happen:
- timestamp nonce transaction that requires multiple signers
- (vacuously true from above) timestamp nonce being used in combination with regular sequence

It seems to be the cleanest, least disruptive way to change `(isd IncrementSequenceDecorator) AnteHandle(` is the above:

Reply:
I believe this will be addressed by the ticket i linked above (https://linear.app/dydx/issue/OTE-624/combine-incrementsequencedecorator-into-sigverificationdecorator)